client/jetpack-connect/index.js

```diff
@@ -72,4 +72,5 @@ export default function() {
 	page( '/jetpack/sso/:siteId?/:ssoNonce?', controller.sso );
 	page( '/jetpack/sso/*', controller.sso );
 	page( '/jetpack/new', controller.newSite );
+	page( '/jetpack/new/*', controller.notFoundError );
```

Comment:
> What happens for /jetpack/new/* will also happen for /jetpack/bla

Yes, I agree, but in the short-term `/jetpack/bla` will still redirect to jetpack.com even after we add `jetpack/new/*` to the whitelist. Let's keep this PR for the `/new` routing problem (which is a current bug), and then plan how we want the route handling to work for all /jetpack/* routes.